react-components/src/components/RuleBasedOutputs/types.ts

```diff
@@ -80,9 +80,10 @@ export type ConcreteExpression = StringExpression | NumericExpression;
 export type Expression = ConcreteExpression | ExpressionOperator;
 
 export type Rule = {
+  type: 'rule';
   id: string; // uuid
   name: string;
-  expression: Expression;
+  expression: Expression | undefined;
```

Comment:
As we've discussed, the deletion of the expression by undefining it simplify the logic created when deleting expressions. to remove the entire Rule when deleting the last concrete expression it would create a too complex code that I dont think we have enough time to work on it right now. I think for now we can keep it and I will work on a better solution later.